test/assertions/to-have-an-item-satisfying.spec.js

```diff
@@ -0,0 +1,122 @@
+/*global expect*/
+describe('to have an item satisfying assertion', function () {
+    it('requires a third argument', function () {
+        expect(
+            function () {
+                expect([1, 2, 3], 'to have an item satisfying');
+            },
+            'to throw',
+            "expected [ 1, 2, 3 ] to have an item satisfying\n" +
+            "  No matching assertion, did you mean:\n" +
+            "  <array-like> to have an item [exhaustively] satisfying <any>\n" +
+            "  <array-like> to have an item [exhaustively] satisfying <assertion>");
+    });
+
+    it('only accepts arrays as the subject', function () {
+        expect(
+            function () {
+                expect(42, 'to have an item satisfying', 'to be a number');
+            },
+            'to throw',
+            "expected 42 to have an item satisfying 'to be a number'\n" +
```

Comment:
Eeek, we should find a way to avoid `'to be a number'` being quoted in the error output. Not sure why that happens here.